File: petpals-pals/src/main/java/com/petpals/pals/domain/pals/model/PalMedicalInformation.java
package com.petpals.pals.domain.pals.model;

import jakarta.validation.constraints.Future;

import java.util.Date;
import java.util.List;

public record PalMedicalInformation(
        boolean isVaccinated,
        List<String> medicalHistory,
        @Future
        Date nextVaccine,
        @Future
        Date nextPlannedVetApp,
        boolean isSterilized
) {
}

File: petpals-pals/src/main/kotlin/com/petpals/domain/model/PalMedicalInformation.kt
package com.petpals.domain.model

import jakarta.validation.constraints.Future
import java.util.*

data class PalMedicalInformation(
    val isVaccinated: Boolean,
    val medicalHistory: List<String>,
    val nextVaccine: @Future Date?,
    val nextPlannedVetApp: @Future Date?,
    val isSterilized: Boolean
)
